src/context/products/productReducer.js

```diff
@@ -1,8 +1,53 @@
 export const reducer = (state, action) => {
-    switch (action.type) {
-      case "FETCH_DATA":
-        return { ...state, products: action.payload.products };
-      default:
-        return state;
+  switch (action.type) {
+
+    case "FETCH_DATA":{
+     const updatedState={
+        ...state,
+        products: action.payload.products,
+        categories: action.payload.categories,
+        selectedCategories: action.payload.categories.reduce((acc, curr) => {
+          return { ...acc, [curr.category_name]: false };
+        }, {}),
+
+        brands:action.payload.brands,
+      selectedBrands: action.payload.brands.reduce((acc, curr) => {
+          return { ...acc, [curr.name]: false };
+        }, {}),
+      };
+      return {...updatedState,intialState:updatedState}
     }
-  };
+
+    case "SORT":
+      console.log(action.payload.value)
+      return { ...state, sortBy: action.payload.value };
+
+    case "BRANDS":
+      let updatedBrands = { ...state.selectedBrands };
+      updatedBrands[action.payload.key] = action.payload.value;
+      return { ...state, selectedBrands: updatedBrands };
+
+    case "INCLUDE_OUT_OF_STOCK":
+      return { ...state, inStock: action.payload.value };
+
+    case "FAST_DELIVERY":
+      return { ...state, fastDelivery: action.payload.value };
+
+    case "RANGE":
+      return { ...state, range: action.payload.value };
+
+    case "RATINGS":
+      return { ...state, ratings: action.payload.value };
+
+    case "CATEGORIES":
+      let updatedCategories = { ...state.selectedCategories };
+      updatedCategories[action.payload.key] = action.payload.value;
+      return { ...state, selectedCategories: updatedCategories };
+
+      case "RESET_FILTER":
+        return {...state.intialState,intialState:state.intialState}
+        
+    default:
+      return state;
+  }
```

Comment:
Suggestion: You can throw an error here instead of just returning the current state

Reply:
for now I have used console log then later I will handle it by  showing alerts .